client.go

```diff
@@ -94,16 +94,33 @@ func newClient(apiKey string, httpClient *http.Client) *Client {
 	}
 }
 
+func validatePathParameters(params []string) error {
+	invalidParams := []string{}
+	for _, param := range params {
+		if len(strings.TrimSpace(param)) == 0 {
+			invalidParams = append(invalidParams, param)
+		}
+	}
+	if len(invalidParams) > 0 {
+		return fmt.Errorf("Parameters cannot be empty strings")
```

Comment:
As of right now, we don't return any generic errors. I'm wondering what the best practice is. in the examples, we don't handle the generic case.

Reply:
Agreed, I wasn't sure what recurly error would be appropriate here. I'm open to thoughts.

Reply:
I suppose that the error Class would be `ErrorClassClient`, but would it make more sense for the error Type to be `ErrorTypeUnknown`, `ErrorTypeBadRequest`, or something else?

Reply:
I don't know. there is no "blowing up" like there is in other languages. maybe we just change the examples.

Reply:
Seems there are actually some other places we do this. so regardless, we will need to consider updating the examples.

Reply:
Nearly every example has to be updated to account for the potential error being returned from the validation anyway, so it's not that much more of a lift